aws/dynamodb_test.go

```diff
@@ -0,0 +1,101 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+
+func createTestDynamoTables(t *testing.T)   {
+	awsSession := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	tables := 0
+	var tablesToCreate []string
+	// Creates 20 tables to simulate a smaller environment. We can make this dynamic if needed.
+	for tables <= 20 {
+
+		tablesToCreate = append(tablesToCreate, "dynamo-nuke-test-" + random.UniqueId() )
+		tables +=1
+
+	}
+
+
+	svc := dynamodb.New(awsSession)
+
+	for num, table := range tablesToCreate {
+		input := &dynamodb.CreateTableInput{
+			TableName: &table,
+			AttributeDefinitions: []*dynamodb.AttributeDefinition{
+				{
+					AttributeName: aws.String("Nuke" + string(rune(num))),
+					AttributeType: aws.String("S"),
+				},
+				{
+					AttributeName: aws.String("TypeofNuke" + string(rune(num))),
+					AttributeType: aws.String("S"),
+				},
+			},
+			KeySchema: []*dynamodb.KeySchemaElement{
+				{
+					AttributeName: aws.String("Nuke" + string(rune(num)) ),
+					KeyType:       aws.String("HASH"),
+				},
+				{
+					AttributeName: aws.String("TypeofNuke" + string(rune(num))),
+					KeyType:       aws.String("RANGE"),
+				},
+			},
+			ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+				ReadCapacityUnits:  aws.Int64(1),
+				WriteCapacityUnits: aws.Int64(1),
+			},
+
+		}
+		_, err := svc.CreateTable(input)
+		time.Sleep(2 * time.Second)
```

Comment:
Why `Sleep`? Is this to account for tables being created async? If so, then we should use a loop to explicitly check if the API knows about the table, rather than sleep, as sleep will fail intermittently.